tycho-apitools-plugin/src/main/java/org/eclipse/tycho/apitools/APIAnalyser.java

```diff
@@ -0,0 +1,92 @@
+package org.eclipse.tycho.apitools;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.pde.api.tools.internal.builder.BaseApiAnalyzer;
+import org.eclipse.pde.api.tools.internal.builder.BuildContext;
+import org.eclipse.pde.api.tools.internal.model.ApiModelFactory;
+import org.eclipse.pde.api.tools.internal.model.BundleComponent;
+import org.eclipse.pde.api.tools.internal.provisional.IApiDescription;
+import org.eclipse.pde.api.tools.internal.provisional.builder.IApiAnalyzer;
+import org.eclipse.pde.api.tools.internal.provisional.model.IApiBaseline;
+import org.eclipse.pde.api.tools.internal.provisional.model.IApiComponent;
+import org.eclipse.pde.api.tools.internal.provisional.problems.IApiProblem;
+import org.eclipse.pde.core.target.ITargetPlatformService;
+import org.eclipse.sisu.equinox.EquinoxServiceFactory;
+
+@Component(role = APIAnalyser.class)
+class APIAnalyser {
+
+	// TODO: or encapsulate this into own class?
+	@Requirement(hint = "connect")
+	private EquinoxServiceFactory serviceFactory;
+
+	public IApiProblem[] analyse(File projectFile, Set<File> projectDependencies, File baselineJar)
+			throws IOException, CoreException {
+		// TODO: See
+		// org.eclipse.pde.api.tools.model.tests.TestSuiteHelper.createTestingBaseline();
+
+		ITargetPlatformService tpService = serviceFactory.getService(ITargetPlatformService.class);
```

Comment:
Be aware that fetching a service can return null, so if you require the service, its better to use Object.requireNonNull(..., "Some usefull message"); this will simplify debugging a lot.